webui/react/src/pages/TrialDetails/TrialChart.tsx

```diff
@@ -65,17 +69,38 @@ const TrialChart: React.FC<Props> = ({
   }, [ metrics, workloads ]);
 
   const chartOptions: Options = useMemo(() => {
+
+    const xScale =
+      chartData[0].length === 1
+        ? {
+          max: chartData[0][0] + A_REASONABLY_SMALL_NUMBER,
+          min: chartData[0][0] - A_REASONABLY_SMALL_NUMBER,
+          time: false,
+        }
+        : { time: false };
+    // const xScale = { time: false };
+
+    const yScale =
+      chartData[1]?.length === 1
+        ? {
+          distr: scale === Scale.Log ? 3 : 1,
+          max: (chartData?.[1][0] ?? 0) + A_REASONABLY_SMALL_NUMBER,
+          min: (chartData?.[1][0] ?? 0) - A_REASONABLY_SMALL_NUMBER,
+        }
+        : { distr: scale === Scale.Log ? 3 : 1 };
+    // const yScale = { distr: scale === Scale.Log ? 3 : 1 };
     return {
       axes: [
         { label: 'Batches' },
         { label: metrics.length === 1 ? getChartMetricLabel(metrics[0]) : 'Metric Value' },
       ],
       height: 400,
+      key: trialId,
```

Comment:
looks like the zoom-preservation code was the culprit here. 

basically if new data comes in with bounds within the previous bound, the `setScale` hook thinks it's a zoom, so we dont reset scales. not sure why data update doesn't happen though.

I added a `key` to our extension of options to allow us to explicitly specify that we have a different chart coming in, and should recreate.

Reply:
**praise:** nice find! it's PRAISE!